sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -103,77 +186,172 @@ def hpat_pandas_series_getitem(self, idx):
     _func_name = 'Operator getitem().'
 
     if not isinstance(self, SeriesType):
-        raise TypingError('{} The object must be a pandas.series. Given: {}'.format(_func_name, self))
+        return None
 
-    if isinstance(idx, types.Integer):
-        def hpat_pandas_series_getitem_idx_integer_impl(self, idx):
-            """
-            **Test**: python -m sdc.runtests sdc.tests.test_series.TestSeries.test_series_iloc1
-            """
+    #note: Getitem return Series
+    index_is_none = self.index is None or isinstance(self.index, numba.types.misc.NoneType)
+    index_is_number = index_is_none or (self.index and isinstance(self.index.dtype, types.Number))
+    index_is_string = not index_is_none and isinstance(self.index.dtype, (types.UnicodeType, types.StringLiteral))
+
+    if (isinstance(idx, types.Number) and index_is_number or
+        (isinstance(idx, (str, types.UnicodeType, types.StringLiteral)) and index_is_string)):
+        def hpat_pandas_series_getitem_index_impl(self, idx):
+            mask = numpy.empty(len(self._data), numpy.bool_)
+            for i in range(len(self.index)):
+                mask[i] = self.index[i] == idx
+            return pandas.Series(self._data[mask], self.index[mask])
+
+        return hpat_pandas_series_getitem_index_impl
 
-            result = self._data[idx]
-            return result
+    if (isinstance(idx, types.Integer) and index_is_string):
+        def hpat_pandas_series_idx_impl(self, idx):
+            return self._data[idx]
 
-        return hpat_pandas_series_getitem_idx_integer_impl
+        return hpat_pandas_series_idx_impl
 
+    #Return slice for str values not implement
     if isinstance(idx, types.SliceType):
         def hpat_pandas_series_getitem_idx_slice_impl(self, idx):
-            """
-            **Test**: python -m sdc.runtests sdc.tests.test_series.TestSeries.test_series_iloc2
-            """
-
-            result = pandas.Series(self._data[idx])
-            return result
+            return pandas.Series(self._data[idx], self.index[idx])
 
         return hpat_pandas_series_getitem_idx_slice_impl
 
+    #Series with str index not implement
     if isinstance(idx, SeriesType):
         def hpat_pandas_series_getitem_idx_series_impl(self, idx):
-            """
-            **Test**: python -m sdc.runtests sdc.tests.test_series.TestSeries.test_setitem_series_bool2
-            """
-            super_index = idx._data
-            result = self._data[super_index]
-            return result
+            res = numpy.copy(self._data[:len(idx._data)])
+            for i in range(len(res)):
+                for j in range(len(self.index)):
```

Comment:
I would say we should use `numba.prange` instead of python `range` to achieve better performance.

Reply:
Fixed